pcdsdevices/epics/areadetector/base.py

```diff
@@ -27,4 +27,12 @@ class ADComponent(base.ADComponent, Component):
 
 
 class ADBase(base.ADBase, Device):
-    pass
+    def stage(self, *args, **kwargs):
+        ret = Device.stage(self, *args, **kwargs)
+        try:
+            self.validate_asyn_ports()
+        except RuntimeError as err:
+            self.unstage(*args, **kwargs)
+            raise err
+        return ret 
```

Comment:
This will soon not be needed as my patch was accepted upstream: https://github.com/NSLS-II/ophyd/pull/415